lib/Page.js

```diff
@@ -168,9 +169,16 @@ Page.prototype.prepareTemplateData = function () {
  */
 Page.prototype.collectHeadings = function (renderedPage) {
   const $ = cheerio.load(renderedPage);
-  $('h1,h2,h3').each((i, heading) => {
-    this.headings[$(heading).attr('id')] = $(heading).text();
-  });
+  let headingsSelector = '';
+  if (this.depthOfHeadings > 0) {
+    let headingsSelector = 'h1';
+    for (let i = 1; i <= this.depthOfHeadings; i += 1) {
+      headingsSelector += `, h${i}`;
+    }
+    $(headingsSelector.slice(0, -2)).each((i, heading) => {
```

Comment:
As mentioned, slice is no longer needed, as the algorithm has changed to no longer include trailing commas.